packages/cosmic-swingset/src/chain-main.js

```diff
@@ -567,29 +530,88 @@ export default async function main(
       afterCommitCallback,
       swingsetConfig,
     });
+  };
+
+  return launchChain;
+};
 
-    const { blockingSend, shutdown } = s;
-    ({ writeSlogObject, savedChainSends } = s);
```

Comment:
I don't see how `savedChainSends` is initialized from the value returned by `launch`. Did that get dropped? The hangover logic is notoriously lacking testing and quite brittle (we have know bugs in it)